### PR TITLE
Stac 12971 centralized authentication (4.4.0)

### PR DESCRIPTION
**Pull request by gggina**

### CODE REVIEW

File: configure/security/rbac/rbac_permissions.md [hunk 1]
```diff
@@ -18,9 +18,10 @@ There are two types of permission in StackState:
 
 ## Predefined roles
 
-StackState comes with three predefined roles:
+StackState comes with four predefined roles:
 
 * **Administrators** \(`stackstate-admin`\): Have all permissions and access to all views.
```
Comment (gggina) [inline configure/security/rbac/rbac_permissions.md:23]: Have access to all views and have all permissions, except for the platform management permission `access-admin-api.

File: configure/security/rbac/rbac_permissions.md [hunk 2]
```diff
@@ -18,9 +18,10 @@ There are two types of permission in StackState:
...
+* **Platform Administrators** \(`stackstate-platform-admin`\): Have platform management permissions.
```
Comment (gggina) [inline configure/security/rbac/rbac_permissions.md:24]: Have platform management permissions and have access to all views.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 3]
```diff
@@ -23,16 +23,19 @@ This page provides specific instructions for upgrading to each currently support
...
-* Authorization configuration for Base API and Admin API has been centralized. This means that there is a single location in the configuration for groups to roles mappings. The three default StackState roles could previously be overridden, but are now always available.
-```
-stackstate {
-  authorization {
-    adminGroups = ${stackstate.authorization.adminGroups} ["custom-admin-role-from-ldap-or-oidc-or-keycloak"]
-    powerUserGroups = ${stackstate.authorization.powerUserGroups} ["custom-power-user-role-from-ldap-or-oidc-or-keycloak"]
-    guestGroups = ${stackstate.authorization.guestGroups} ["custom-guest-role-from-ldap-or-oidc-or-keycloak"]
+* Authentication and Authorization configurations for Base API and Admin API have been centralized. This means there is single configuration for groups to roles mappings and single authentication provider used for both Base and Admin API. The three default StackState roles could previously be overridden - `stackstate-admin`, `stackstate-power-user`, `stackstate-guest`, but are now always available (additionally new default role `stackstate-platform-admin` has been introduced).
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:26]: * Authentication and Authorization configurations for the Base API and Admin API have been centralized. This means there is a single configuration for groups to roles mappings and a single authentication provider used for both the Base API and Admin API. The default StackState roles are now always available, these could previously be overridden - `stackstate-admin`, `stackstate-power-user`, `stackstate-guest`. Additionally, a new default role `stackstate-platform-admin` has been introduced.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 4]
```diff
@@ -23,16 +23,19 @@ This page provides specific instructions for upgrading to each currently support
...
+      adminGroups = ${stackstate.authorization.adminGroups} ["custom-admin-role-from-ldap-or-oidc-or-keycloak"]
+      platformAdminGroups = ${stackstate.authorization.platformAdminGroups} ["custom-platform-admin-role-from-ldap-or-oidc-or-keycloak"]
+      powerUserGroups = ${stackstate.authorization.powerUserGroups} ["custom-power-user-role-from-ldap-or-oidc-or-keycloak"]
+      guestGroups = ${stackstate.authorization.guestGroups} ["custom-guest-role-from-ldap-or-oidc-or-keycloak"]    
+    }
   }
-}
-```
+  ```
+  Note, that the platform management and platform content management permissions have been separated into two groups - `adminGroup` and `platformAdminGroup`.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:37]: Platform management and platform content management permissions have been separated into two groups - `platformAdminGroup` and `adminGroup`. Users in the group platformAdminGroup are limited to only platform management tasks, such as change database retention, clear database, clear caches and view logs. Users in the group adminGroup no longer have platform management permissions.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 5]
```diff
@@ -23,16 +23,19 @@ This page provides specific instructions for upgrading to each currently support
...
+  The users that are in `platformAdminGroup` are limited to only platform management tasks such as - change database retention, clear database, clear caches, view logs etc. Correspondingly, users in `adminGroup` cannot do platform management anymore. If you are using LDAP, Keycloak or OIDC auth provider, you have to configure auth provider with a separate role/group for platform management and assign the role to a limited number of users. If you wish one user to manage the content and the platform, then you still need to configure auth provider with two separate roles/groups and then assign those to a single user in the settings of the auth provider. You should not map the same provider role/group to different stackstate authorization groups.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:38]: If you are using LDAP, Keycloak or an OIDC auth provider, you will have to configure the auth provider with a separate role/group for platform management and assign the role to a limited number of users. If you wish for one user to manage both the content and the platform, you will still need to configure the auth provider with two separate roles/groups and then assign both of those to a single user in the settings of the auth provider. You should not map the same provider role/group to different StackState authorization groups.

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 6]
```diff
@@ -68,16 +72,19 @@ stackstate {
...
-* Authorization configuration for Base API and Admin API has been centralized. This means that there is a single location in the configuration for groups to roles mappings. The three default StackState roles could previously be overridden, but are now always available.
-```
-stackstate {
-  authorization {
-    adminGroups = ${stackstate.authorization.adminGroups} ["custom-admin-role-from-ldap-or-oidc-or-keycloak"]
-    powerUserGroups = ${stackstate.authorization.powerUserGroups} ["custom-power-user-role-from-ldap-or-oidc-or-keycloak"]
-    guestGroups = ${stackstate.authorization.guestGroups} ["custom-guest-role-from-ldap-or-oidc-or-keycloak"]
+* Authentication and Authorization configurations for Base API and Admin API have been centralized. This means there is single configuration for groups to roles mappings and single authentication provider used for both Base and Admin API. The three default StackState roles could previously be overridden - `stackstate-admin`, `stackstate-power-user`, `stackstate-guest`, but are now always available (additionally new default role `stackstate-platform-admin` has been introduced).
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:75]: as above

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 7]
```diff
@@ -68,16 +72,19 @@ stackstate {
...
+      adminGroups = ${stackstate.authorization.adminGroups} ["custom-admin-role-from-ldap-or-oidc-or-keycloak"]
+      platformAdminGroups = ${stackstate.authorization.platformAdminGroups} ["custom-platform-admin-role-from-ldap-or-oidc-or-keycloak"]
+      powerUserGroups = ${stackstate.authorization.powerUserGroups} ["custom-power-user-role-from-ldap-or-oidc-or-keycloak"]
+      guestGroups = ${stackstate.authorization.guestGroups} ["custom-guest-role-from-ldap-or-oidc-or-keycloak"]    
+    }
   }
-}
-```
+  ```
+  Note, that the platform management and platform content management permissions have been separated into two groups - `adminGroup` and `platformAdminGroup`.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:86]: as above

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 8]
```diff
@@ -68,16 +72,19 @@ stackstate {
...
+  The users that are in `platformAdminGroup` are limited to only platform management tasks such as - change database retention, clear database, clear caches, view logs etc. Correspondingly, users in `adminGroup` cannot do platform management anymore. If you are using LDAP, Keycloak or OIDC auth provider, you have to configure auth provider with a separate role/group for platform management and assign the role to a limited number of users. If you wish one user to manage the content and the platform, then you still need to configure auth provider with two separate roles/groups and then assign those to a single user in the settings of the auth provider. You should not map the same provider role/group to different stackstate authorization groups.
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:87]: as above

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md [hunk 9]
```diff
@@ -87,7 +94,7 @@ stackstate {
       authentication {
         ...
         adminGroups = ["your-custom-oidc-or-ldap-or-keycloak-admin-role"]
-        powerUserGroups = ["your-custom-oidc-or-ldap-or-keycloak-power-user-role"]
+        platformAdminGroups = ["your-custom-oidc-or-ldap-or-keycloak-platform-admin-role"]
```
Comment (gggina) [inline setup/upgrade-stackstate/version-specific-upgrade-instructions.md:97]: should the powerUserGroups have been removed?

File: configure/security/authentication/authentication_options.md [hunk 10]
```diff
@@ -16,13 +16,14 @@ For better security StackState can be configured to use exactly one of the follo
 
 ## User roles
 
-When a user has been authenticated permissions for that user are usually assigned based of the roles the user has. The documentation for the specific authentication mechanisms also contain examples on how to map the roles or groups from the external systems to the 3 standard roles of StackState:
+When a user has been authenticated permissions for that user are usually assigned based of the roles the user has. The documentation for the specific authentication mechanisms also contain examples on how to map the roles or groups from the external systems to the 4 standard roles of StackState:
 
 * **Guest** - able to see information but make no changes.
 * **Power User** - able to see and change all configuration and install StackPacks.
-* **Administrator** - able to see and change all configuration, install StackPacks, grant and revoke user permissions and upload \(new versions of\) StackPacks.
+* **Administrator** - able to see and change content of StackState, e.g. all configuration, install StackPacks, grant and revoke user permissions and upload \(new versions of\) StackPacks.
```
Comment (gggina) [inline configure/security/authentication/authentication_options.md:23]: able to see and change content of StackState. For example, see all configuration, install StackPacks, grant and revoke user permissions and upload \(new versions of\) StackPacks.

File: configure/security/authentication/authentication_options.md [hunk 11]
```diff
@@ -16,13 +16,14 @@ For better security StackState can be configured to use exactly one of the follo
...
+* **Platform Administrator** - able to perform management of the StackState platform, e.g. change data retention, clear database, view logs, cache management.
```
Comment (gggina) [inline configure/security/authentication/authentication_options.md:24]: able to perform management of the StackState platform. For example, change data retention, clear the database, view logs and cache management.

File: configure/security/authentication/authentication_options.md [hunk 12]
```diff
@@ -16,13 +16,14 @@ For better security StackState can be configured to use exactly one of the follo
...
 
-When deciding on the roles to assign your users it is strongly advised to have only a small group of Administrators, for example only the engineers responsible for installing StackState and doing the initial configuration. Administrator users can manage access to StackState and decide which StackPacks can be used. Installing StackPacks and other fine tuning of the configuration can be delegated to a larger number of users with the Power User role.
+When deciding on the roles to assign your users it is strongly advised to have only a small group of Platform Administrators and Administrators, for example only the engineers responsible for installing StackState and doing the initial configuration. Administrator users can manage access to StackState and decide which StackPacks can be used. Installing StackPacks and other fine tuning of the configuration can be delegated to a larger number of users with the Power User role.
```
Comment (gggina) [inline configure/security/authentication/authentication_options.md:26]: When deciding on the roles to assign your users, it is strongly advised to have only a small group of Platform Administrators and Administrators. For example, only the engineers responsible for installing StackState and doing the initial configuration. Administrator users can manage access to StackState and decide which StackPacks can be used. Installing StackPacks and other fine tuning of the configuration can be delegated to a larger number of users with the Power User role.

File: configure/security/authentication/authentication_options.md [hunk 13]
```diff
@@ -16,13 +16,14 @@ For better security StackState can be configured to use exactly one of the follo
...
+Platform Administrator users can clear database, change data retention, view logs and perform other platform management tasks.
```
Comment (gggina) [inline configure/security/authentication/authentication_options.md:27]: Platform Administrator users can clear the database, change data retention settings, view logs and perform other platform management tasks.

File: configure/security/rbac/rbac_permissions.md [hunk 14]
```diff
@@ -104,46 +107,46 @@ System permissions scope user capabilities, such as access to settings, query ex
...
 See also the full list of [system permissions](rbac_permissions.md#system-permissions).
 
-| Permission | Purpose | Guest | Power user | Admin |
-| :--- | :--- | :---: | :---: | :---: |
-| `access-view` | Access a specific view \(when granted on a view\) or all views \(when granted on `everything`\). | ✅ <br />`everything` | ✅ <br />`everything` | ✅ <br />`everything` |
-| `delete-view` | Delete a specific view \(when granted on a view\) or all views \(when granted on `everything`\). | - | ✅ <br />`everything` | ✅ <br />`everything` |
-| `save-view` | Update a specific view \(when granted on a view\) or all views \(when granted on `everything`\). | - | ✅ <br />`everything` | ✅ <br />`everything` |
+| Permission | Purpose | Guest | Power user | Admin | Platform Admin |
+| :--- | :--- | :---: | :---: | :---: | :---: |
+| `access-view` | Access a specific view \(when granted on a view\) or all views \(when granted on `everything`\). | ✅ <br />`everything` | ✅ <br />`everything` | ✅ <br />`everything` | ✅ <br />`everything` |
```
Comment (gggina) [inline configure/security/rbac/rbac_permissions.md:147]: Just to confirm, does the platform admin have the `access-view` permission?

File: configure/security/rbac/rbac_permissions.md (hunk 1 above)
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:23]: fixed

File: configure/security/rbac/rbac_permissions.md (hunk 2 above)
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:24]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 5 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:38]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 6 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:75]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 7 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:86]: fixed

File: setup/upgrade-stackstate/version-specific-upgrade-instructions.md (hunk 9 above)
Comment (konstantintatarnikov) [reply setup/upgrade-stackstate/version-specific-upgrade-instructions.md:97]: this is mistake, fixing...

File: configure/security/authentication/authentication_options.md (hunk 10 above)
Comment (konstantintatarnikov) [reply configure/security/authentication/authentication_options.md:23]: fixed

File: configure/security/authentication/authentication_options.md (hunk 11 above)
Comment (konstantintatarnikov) [reply configure/security/authentication/authentication_options.md:24]: fixed

File: configure/security/authentication/authentication_options.md (hunk 13 above)
Comment (konstantintatarnikov) [reply configure/security/authentication/authentication_options.md:27]: fixed

File: configure/security/rbac/rbac_permissions.md (hunk 14 above)
Comment (konstantintatarnikov) [reply configure/security/rbac/rbac_permissions.md:147]: yes, it has

File: configure/security/authentication/authentication_options.md (hunk 12 above)
Comment (konstantintatarnikov) [reply configure/security/authentication/authentication_options.md]: fixed

**Pull request closed without merging**